pkg/solana/logpoller/log_poller.go

```diff
@@ -282,31 +287,29 @@ func (lp *Service) ReplayStatus() ReplayStatus {
 	return lp.replay.status
 }
 
-func (lp *Service) getLastProcessedSlot(ctx context.Context) (int64, error) {
-	if lp.lastProcessedSlot != 0 {
-		return lp.lastProcessedSlot, nil
-	}
-
-	latestDBBlock, err := lp.orm.GetLatestBlock(ctx)
-	if err == nil {
-		return latestDBBlock, nil
+func (lp *Service) getLastProcessedSlot(ctx context.Context) (lastProcessed int64, err error) {
+	lastProcessed = lp.lastProcessedSlot
+	if lastProcessed == 0 {
+		lastProcessed, err = lp.orm.GetLatestBlock(ctx)
 	}
 
-	if !errors.Is(err, sql.ErrNoRows) {
-		return 0, fmt.Errorf("error getting latest block from db: %w", err)
+	if err != nil {
+		if !errors.Is(err, sql.ErrNoRows) {
+			return 0, fmt.Errorf("error getting latest block from db: %w", err)
+		}
+		lastProcessed = 0
 	}
 
-	latestFinalizedSlot, err := lp.client.SlotHeightWithCommitment(ctx, rpc.CommitmentFinalized)
+	var lookbackBlock int64
+	lookbackBlock, err = lp.computeLookbackWindow(ctx)
```

Comment:
Do we need to calculate lookbackBlock on every iteration?

Reply:
Ok, I'll add a cache for it

Reply:
Actually, I'm not sure there's much we could do. We have to find the latest finalized block and first available slot from the chain each time, because we have no idea how much either of those may have changed since last time. It's returning a number that just combines those results with MaxRetention... which is already cached.

Did you have in mind a way we could improve this?

Reply:
I guess we could estimate what the present block number is, based on the block time and elapsed time since the last call?

Reply:
> If someone takes down the testnet rpc servers for a month and then brings them back up
I don't think that's a practical scenario we should think ok.

Imo, lets keep it simple. Lookback is only for the time when the Node is restarted, or brought up first time. So good to do it at startup once, and not repeatedly.

Reply:
Yes I think it should be fine.